### PR TITLE
Migration to hapi v17

### PR DESCRIPTION
**Pull request by guischdi**
Hey Brandon,
I ported your library for hapi version 17 compatibility. Also adjusted the tests and everything works fine for me in a hapi v17 project.
Pleas keep in mind to release a new major version for backwards compatibility reasons. Thx!

### CODE REVIEW

**Comment by PizzaBrandon** [comment]
Thanks for the PR. I hope to have it reviewed within a few days.

**Comment by guischdi** [comment]
Alright. Thanks a lot!

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,7 @@
 language: node_js
 
 node_js:
-  - 6
-  - 4
-  - "node"
+  - 8
+  - 9
```
Comment (PizzaBrandon) [inline .travis.yml]: I don't want to imply support for non-LTS versions. Please remove this.

File: src/dogear.js [hunk 2]
```diff
@@ -5,141 +5,133 @@ const Oppsy = require('oppsy');
...
 const At = require('lodash.at');
 
-const internals = {
-  defaults: {
-    statsdConfig: {
-      prefix: 'hapi.'
-    },
-    opsInterval: 1000,
-    opsMetrics: null
+const defaultConfig = {
```
Comment (PizzaBrandon) [inline src/dogear.js]: This is a non-functional styling change that adds unnecessarily to the diff. There is no perceivable benefit to this change. Please revert and limit changes to the scope of the change, and not introduce different structure and styling without functional benefit.

File: src/dogear.js [hunk 3]
```diff
@@ -5,141 +5,133 @@ const Oppsy = require('oppsy');
...
-    self.client.histogram('request.response_time', msec, tags);
+        this.client.increment(`request.status.${ statusCode }`, 1, tags);
+        this.client.increment('request.received', 1, tags);
+      }
 
-    reply.continue();
-  });
+      this.client.histogram('request.response_time', msec, tags);
 
+      return h.continue;
```
Comment (PizzaBrandon) [inline src/dogear.js]: This appears to be the only functional change in this entire file. Please revert all other changes and leave this line as `return reply.continue;`

File: package-lock.json [hunk 4]
```diff
@@ -0,0 +1,4116 @@
+{
```
Comment (PizzaBrandon) [inline package-lock.json]: Please remove this file and add it to `.gitignore`. The library was developed prior to NPM 5's wide-spread use and wasn't prepared to block it already.

File: .eslintrc [hunk 5]
```diff
@@ -6,10 +6,6 @@
     "es6": true
   },
 
-  "ecmaFeatures": {
```
Comment (PizzaBrandon) [inline .eslintrc]: Unrelated to functional change.

File: package.json [hunk 6]
```diff
@@ -1,56 +1,56 @@
 {
-	"name": "dogear",
-	"description": "A hapi plugin for sending expanded metrics to statsd-compliant services",
-	"version": "2.0.3",
-	"engines": {
-		"node": ">=4.0.0"
-	},
-	"main": "src/index.js",
-	"scripts": {
+  "name": "dogear",
```
Comment (PizzaBrandon) [inline package.json]: There was no benefit to this file to restructure it and cause all the lines to be changed. Please do not explode the diff for the sake of changing style; limit changes to just what needs to be changed to enable the new functionality. Please revert these changes and leave only the relevant package and node support changes.

File: test/dogear.test.js [hunk 7]
```diff
@@ -243,82 +240,82 @@ describe('Plugin reports on paths.', (tap) => {
...
-        opsInterval: 500
+        opsInterval: 10
       }
-    })
-    .then(() => {
+    });
 
-      const client = server.statsd;
+    const client = server.statsd;
+    const gaugeSpy = sandbox.spy(client, 'gauge');
```
Comment (PizzaBrandon) [inline test/dogear.test.js:283]: I believe a stub should be safer here to prevent the client from attempting to do work. We just want to verify the gauge was invoked.

File: test/dogear.test.js [hunk 8]
```diff
@@ -243,82 +240,82 @@ describe('Plugin reports on paths.', (tap) => {
...
-    server.register({
-      register: plugin,
+    await server.register({
+      plugin,
       options: {
         statsdConfig: {
           mock: true
         },
         opsInterval: 500,
-        opsMetrics: [ 'bad.op.metric' ]
```
Comment (PizzaBrandon) [inline test/dogear.test.js]: Unnecessary styling change.

File: test/dogear.test.js [hunk 9]
```diff
@@ -105,7 +102,7 @@ describe('Plugin reports on paths.', (tap) => {
   });
 
 
-  tap.test('Reports stats for request.', (t) => {
+  tap.test('Reports stats for request.', async t => {
```
Comment (PizzaBrandon) [inline test/dogear.test.js]: For clarity of code, I prefer that all arrow functions have parenthesis around their parameters. Please change these to `async (t) =>`.

File: src/dogear.js (hunk 2 above)
Comment (PizzaBrandon) [reply src/dogear.js]: `hapi` did change their default code style, but I prefer that PRs are limited to correcting only the issue at hand. The current style of this file will still work with newer versions of `hapi`, thus no pressing reason to change it now.

**Review by PizzaBrandon — CHANGES_REQUESTED** [review]

File: package.json (hunk 6 above)
Comment (guischdi) [reply package.json]: So you really want mixed tabs and spaces?
Anyway I changed this, but `npm` just re-arranged the order of the dependencies...

File: test/dogear.test.js (hunk 9 above)
Comment (guischdi) [reply test/dogear.test.js]: I changed this. But could you please adjust your linter rules if like a certain style more than another?

File: test/dogear.test.js (hunk 7 above)
Comment (guischdi) [reply test/dogear.test.js:283]: Sinon stubs with the signature `sinon.stub(object, "method", func)` have been deprecated in v3.0.0 of sinon. I also tried `stub.callsFake(fakeFunction)` but this won't work with async function. And the only thing we want to achieve here, is to verify the gauge was invoked (as you mentioned). We don't need to replace the whole function for this. Btw: Where should the client send anything when there's no statsd server speciefied anywhere in the whole repository.

File: test/dogear.test.js (hunk 8 above)
Comment (guischdi) [reply test/dogear.test.js]: Please prevent this by eslint rules, so you won't have to tell everybody who's creating a PR for you. Changed it anyway.

**Review by guischdi — COMMENTED** [review]
I reversed the style changes. Could you please check this again? Thx!

**Comment by guischdi** [comment]
@PizzaBrandon, any news about this PR?

**Comment by PizzaBrandon** [comment]
Things are really busy right now. I hope to have a chance to review this week.

**Comment by PizzaBrandon** [comment]
Thanks for helping update `dogear` to support `hapi` 17

**Comment by guischdi** [comment]
Thanks for merging, @PizzaBrandon !
